app/controller/grid/Grid.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * This class is the controller for the cpsi mapview WFS
+ * generic grid class
+ *
+ */
+Ext.define('CpsiMapview.controller.grid.Grid', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.cmv_grid',
+
+    requires: [
+        'Ext.menu.Menu',
+        'Ext.grid.filters.Filters',
+        'BasiGX.util.Layer',
+        'GeoExt.util.OGCFilter'
+    ],
+
+    /**
+    * The currently active spatial filter for the layer.
+    *
+    * @cfg {Ext.util.Filter} spatialFilter
+    */
+    spatialFilter: null,
+
+    /**
+     * Fire a onRowDblClick to be handled by the parent application
+     */
+    onRowDblClick: function (sender, record) {
+        this.fireEvent('onRowDblClick', sender, record);
+    },
+
+    /**
+     * Zoom the map to the selected feature with a buffer
+     *
+     * @param {ol.Feature} feature
+     * @param {ol.Map} map
+     * @private
+     */
+    zoomToFeature: function (feature, map) {
+
+        // TODO check for feature type when zooming
+        var extent = feature.getGeometry().getExtent();
+        // as this is a point then buffer it by 100m
+        extent = ol.extent.buffer(extent, 100);
```

Comment:
Making the buffer width configurable in the grid would be a good idea, IMHO.

Reply:
Added as config option